packages/profile-sync-controller/src/controllers/user-storage/utils.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Returns the difference between 2 sets.
+ * NOTE - this is temporary until we can support Set().difference method
```

Comment:
Aww, the original implementation used the new JS [Set utilities](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/difference).

However both our TS version, and transpilers did not play well with it.

Still kept the Set data structure (a bit nicer to work with) and remade these utilities.